MQ2AutoSize.cpp

```diff
@@ -359,269 +536,1087 @@ PLUGIN_API void OnPulse()
+	}
+	else if (selectedComms == static_cast<int>(CommunicationMode::EQBC))
+	{
+		if (who == "group")
+			groupCommand += fmt::format("/bcg /{}", instruction);
+		else if (who == "all")
+			groupCommand += fmt::format("/bca /{}", instruction);
 	}
+
+	if (!groupCommand.empty())
+		DoCommandf(groupCommand.c_str());
 }
 
-PLUGIN_API void InitializePlugin()
+/**
+ * This function checks the loaded state of the MQ2EQBC and MQ2Dannet plugins
+ * and sets the communication mode accordingly.
+ * 
+ * - If DanNet is loaded, it sets the communication mode to DanNet.
+ * - If EQBC is loaded and DanNet is not loaded, it sets the communication mode to EQBC.
+ * - If neither plugin is loaded, it sets the communication mode to None.
+ *
+ * It also handles transitions between different plugin states to ensure that
+ * the communication mode is updated correctly when the state changes.
+ * 
+ * If both communication plugins are loaded, the default is to use DanNet.
+ */
+// choose the plugin to use for communication
+void ChooseInstructionPlugin()
 {
-	EzDetour(PlayerZoneClient__ChangeHeight, &PlayerZoneClient_Hook::ChangeHeight_Detour, &PlayerZoneClient_Hook::ChangeHeight_Trampoline);
+	bool prev_loaded_eqbc = loaded_eqbc;
+	bool prev_loaded_dannet = loaded_dannet;
 
-	AddCommand("/autosize", AutoSizeCmd);
-	LoadINI();
+	loaded_eqbc = GetPlugin("MQ2EQBC");
+	loaded_dannet = GetPlugin("MQ2Dannet");
+
+	if (loaded_dannet)
+	{
+		selectedComms = static_cast<int>(CommunicationMode::DanNet);
+	}
+	else if (loaded_eqbc)
+	{
+		selectedComms = static_cast<int>(CommunicationMode::EQBC);
+	}
+	else
+	{
+		selectedComms = static_cast<int>(CommunicationMode::None);
+	}
+
+	// check for changes in loaded plugins
+	if (prev_loaded_dannet && !loaded_dannet && loaded_eqbc)
+	{
+		selectedComms = static_cast<int>(CommunicationMode::EQBC);
+	}
+	else if (prev_loaded_eqbc && !loaded_eqbc && loaded_dannet)
+	{
+		selectedComms = static_cast<int>(CommunicationMode::DanNet);
+	}
+	else if (!prev_loaded_eqbc && !prev_loaded_dannet)
+	{
+		if (loaded_eqbc && !loaded_dannet)
+		{
+			selectedComms = static_cast<int>(CommunicationMode::EQBC);
+		}
+		else if (!loaded_eqbc && loaded_dannet)
+		{
+			selectedComms = static_cast<int>(CommunicationMode::DanNet);
+		}
+		else
+		{
+			selectedComms = static_cast<int>(CommunicationMode::None);
+		}
+	}
 }
 
-PLUGIN_API void ShutdownPlugin()
+/**
+ * This function adjusts the configuration settings based on the provided type.
+ * 
+ * It can emulate a "zonewide" configuration by setting a large range. 
+ * It can also revert to a previous range value for "range" type.
+ *
+ * @param type The type of configuration to emulate. Valid values are "zonewide" and "range".
+ *    - "zonewide": Sets the range to FAR_CLIP_PLANE value.
+ *    - "range": Sets the range to the previously configured value.
+ */
+// this function is used as a toggle between Zone and Range
+// params: zonewide or range
+void emulate(const std::string& type)
```

Comment:
you only ever pass string literals to this function consider using std::string_view instead.